contracts/amm/MinterAmm.sol

```diff
@@ -876,13 +978,46 @@ contract MinterAmm is
         seriesController.closePosition(seriesId, closeAmount);
 
         // Send the tokens to the seller
-        collateralToken.safeTransfer(msg.sender, collateralAmount);
+        collateralToken.safeTransfer(msg.sender, collateralAmount - tradeFee);
+
+        // If fees were taken, move them to the destination
+        if (tradeFee > 0) {
+            collateralToken.safeTransfer(feeDestinationAddress, tradeFee);
+            emit TradeFeesPaid(feeDestinationAddress, tradeFee);
+        }
 
         // Emit the event
-        emit BTokensSold(msg.sender, seriesId, bTokenAmount, collateralAmount);
+        emit BTokensSold(
+            msg.sender,
+            seriesId,
+            bTokenAmount,
+            collateralAmount - tradeFee
+        );
 
         // Return the amount of collateral received during sale
-        return collateralAmount;
+        return collateralAmount - tradeFee;
+    }
+
+    /// @notice Calculate premium (i.e. the option price) to buy bTokenAmount bTokens for the
+    /// given Series without including any trade fees
+    /// @notice The premium depends on the amount of collateral token in the pool, the reserves
+    /// of bToken and wToken in the pool, and the current series price of the underlying
+    /// @param seriesId The ID of the Series to buy bToken on
+    /// @param bTokenAmount The amount of bToken to buy, which uses the same decimals as
+    /// the underlying ERC20 token
+    /// @return The amount of collateral token necessary to buy bTokenAmount worth of bTokens
+    function bTokenGetCollateralInWithoutFees(
```

Comment:
Would that be easier / more elegant if `bTokenGetCollateralIn(Out)` methods returned 2 values: `(uint collateralBeforeFee, uint fee) = this.bTokenGetCollateralIn(...)`? Did you consider this approach?

Reply:
Returning 2 values would be more elegant, but it would make it more difficult for clients, who now need to differentiate between old AMM's which only return a single value, and the new AMM's which return 2 values. It complicates the smart contract logic so the clients can be simpler.

Reply:
Yeah, backwards compatibility was the reason to go with this approach.   If we don't care about that then the 2 values might be better.

Reply:
Yeah, backwards compatibility was the reason to go with this approach.  We can update it if we want though.